src/Microsoft.DotNet.Interactive.CSharpProject/RoslynWorkspaceUtilities/RoslynWorkspaceUtilities.cs

```diff
@@ -0,0 +1,332 @@
+        if (!TryGetSupportedLanguageName(buildProjectData.ProjectFilePath, out string languageName))
+        {
+            return null;
+        }
+        return ProjectInfo.Create(
+            projectId,
+            VersionStamp.Create(),
+            projectName,
+            projectName,
+            languageName,
+            filePath: buildProjectData.ProjectFilePath,
+            outputFilePath: buildProjectData.TargetPath,
+            documents: GetDocuments(buildProjectData, projectId),
+            projectReferences: GetExistingProjectReferences(buildProjectData, workspace),
+            metadataReferences: GetMetadataReferences(buildProjectData),
+            analyzerReferences: GetAnalyzerReferences(buildProjectData, workspace),
+            parseOptions: CreateParseOptions(buildProjectData, languageName),
+            compilationOptions: CreateCompilationOptions(buildProjectData, languageName));
+    }
+
+    private static bool TryGetSupportedLanguageName(string projectPath, out string languageName)
+    {
+        switch (Path.GetExtension(projectPath))
+        {
+            case ".csproj":
+                languageName = LanguageNames.CSharp;
+                return true;
+            default:
+                languageName = null;
+                return false;
+        }
+    }
+
+    private static IReadOnlyList<DocumentInfo> GetDocuments(BuildProjectData buildProjectData, ProjectId projectId) =>
+        buildProjectData
+            .SourceFiles?
+                .Where(File.Exists)
+                .Select(x => DocumentInfo.Create(
+                    DocumentId.CreateNewId(projectId),
+                    Path.GetFileName(x),
+                    loader: TextLoader.From(
+                        TextAndVersion.Create(
+                        SourceText.From(File.ReadAllText(x), Encoding.Unicode), VersionStamp.Create())),
+                    filePath: x)).ToList();
+
+    private static IReadOnlyList<ProjectReference> GetExistingProjectReferences(BuildProjectData buildProjectData, CodeAnalysis.Workspace workspace) =>
+        buildProjectData.ProjectReferences
+            .Select(x => workspace.CurrentSolution.Projects.FirstOrDefault(y => y.FilePath.Equals(x, StringComparison.OrdinalIgnoreCase)))
+            .Where(x => x != null)
+            .Select(x => new ProjectReference(x.Id)).ToList();
+
+    private static IReadOnlyList<MetadataReference> GetMetadataReferences(BuildProjectData buildProjectData) =>
+        buildProjectData
+            .References?.Where(File.Exists)
+            .Select(x => MetadataReference.CreateFromFile(x)).ToList();
+
+    private static IReadOnlyList<AnalyzerReference> GetAnalyzerReferences(BuildProjectData buildProjectData, CodeAnalysis.Workspace workspace)
+    {
+        IAnalyzerAssemblyLoader loader = workspace.Services.GetRequiredService<IAnalyzerService>().GetLoader();
+
+        return buildProjectData
+                .AnalyzerReferences?
```

Comment:
This method should not be allowed to return null. If there are no source files, we should return an empty list.

Reply:
fixed

Reply:
fixed